### PR TITLE
refactor ThreadSampling env vars

### PR DESCRIPTION
**Pull request by Kielek**
## Why

Cleanup scaffolding for Thread Sampling.
Values needed for exporter.

## What

Refactor reading env. variables for Thread Sampling

## Tests

Manual test for Console application. Still working.

### CODE REVIEW

File: tracer/src/Datadog.Trace/ClrProfiler/Instrumentation.cs [hunk 1]
```diff
@@ -180,9 +181,7 @@ public static void Initialize()
             // Thread Sampling ("profiling") feature
             try
             {
-                // If you change this, change environment_variables.h too
-                var enabled = EnvironmentHelpers.GetEnvironmentVariable("SIGNALFX_THREAD_SAMPLING_ENABLED", "false");
-                if (enabled != null && (enabled.ToLower() == "1" || enabled.ToLower() == "true"))
+                if (TracerManager.Instance.Settings.ThreadSamplingEnabled)
```
Comment (pellared) [inline tracer/src/Datadog.Trace/ClrProfiler/Instrumentation.cs:184]: you should be able to pass this value through the constructor in `TracerManagerFactory.CreateTracerManager()`
`

**Review by pellared — COMMENTED** [review]
just comments to reduce the "globals" explosion 😄 
apart from it LGTM

File: tracer/src/Datadog.Trace/AsyncLocalScopeManager.cs [hunk 2]
```diff
@@ -12,20 +13,12 @@ namespace Datadog.Trace
...
-            var enabled = Environment.GetEnvironmentVariable("SIGNALFX_THREAD_SAMPLING_ENABLED");
-            if (enabled != null && (enabled.ToLower() == "1" || enabled.ToLower() == "true"))
-            {
-                PushScopeToNative = true;
-            }
-            else
-            {
-                PushScopeToNative = false;
-            }
+            PushScopeToNative = TracerManager.Instance.Settings.ThreadSamplingEnabled;
```
Comment (pellared) [inline tracer/src/Datadog.Trace/AsyncLocalScopeManager.cs:21]: you should be able to pass this value through the constructor in `TracerManagerFactory.CreateTracerManager()`

File: tracer/src/Datadog.Trace/ThreadSampling/ThreadSampler.cs [hunk 3]
```diff
@@ -53,7 +27,7 @@ private static void ReadOneSample(byte[] buf)
 
         private static void SampleReadingThread()
         {
-            int period = GetThreadSamplingPeriod();
+            int period = TracerManager.Instance.Settings.ThreadSamplingPeriod;
```
Comment (pellared) [inline tracer/src/Datadog.Trace/ThreadSampling/ThreadSampler.cs:30]: you should be able to pass this value from `Instumentation.Initialize()` via  `ThreadSampler.Initialize()`

File: tracer/src/Datadog.Trace/AsyncLocalScopeManager.cs (hunk 2 above)
Comment (Kielek) [reply tracer/src/Datadog.Trace/AsyncLocalScopeManager.cs]: fixed

File: tracer/src/Datadog.Trace/ThreadSampling/ThreadSampler.cs (hunk 3 above)
Comment (Kielek) [reply tracer/src/Datadog.Trace/ThreadSampling/ThreadSampler.cs]: fixed

**Review by pellared — APPROVED** [review]

**Pull request merged**